### PR TITLE
Corrected discussion about threading in .NET

### PR DESCRIPTION
**Pull request by papafe**
I'm not sure if we want to use this exact wording/structure, but I hope the sentiment of the PR is clear.

### CODE REVIEW

File: source/sdk/dotnet/crud/threading.txt [hunk 1]
```diff
@@ -52,16 +52,16 @@ Don't lock to read:
...
-  each other. A write on the UI thread may result in your
+  writer at a time. Consequently, synchronous write transactions block
+  each other. A synchronous write on the UI thread may result in your
   app appearing unresponsive while it waits for a write on a
   background thread to complete. If you are using
-  :ref:`Device Sync <sync>`, avoid writing on
+  :ref:`Device Sync <sync>`, avoid synchronous writes on
   the UI thread as Sync writes on a background
-  thread.
+  thread. Prefer to use asynchronous writes in this case, with `realm.WrireAsync`.
```
Comment (MongoCaleb) [inline source/sdk/dotnet/crud/threading.txt:64]: @papafe please change the newly-added sentence to:

> Instead, use ``realm.WriteAsync``.

File: source/sdk/dotnet/sync/add-sync-to-app.txt [hunk 2]
```diff
@@ -59,10 +59,10 @@ The following code creates a new ``Task`` object and writes it to the realm:
...
+.. important:: When Using Sync, Avoid Synchronous Writes on the Main Thread
 
    The fact that Realm performs sync integrations on a background thread means
-   that if you write to your realm on the main thread, there's a small chance your UI
+   that if you do a synchronous write to your realm on the main thread, there's a small chance your UI
    could appear to hang as it waits for the background sync thread to finish a write
-   transaction. Therefore, it's a best practice :ref:`not to write on the main thread
-   when using Device Sync <dotnet-threading-three-rules>`.
+   transaction. Therefore, it's a best practice :ref:`not to do a synchronous write on the main thread
+   when using Device Sync <dotnet-threading-three-rules>` and instead use asynchronous writes with `realm.WriteAsync`.
```
Comment (MongoCaleb) [inline source/sdk/dotnet/sync/add-sync-to-app.txt:68]: @papafe please wrap realm.WriteAsync in two ticks: ``realm.WriteAsync``

**Review by MongoCaleb — COMMENTED** [review]
2 fixes, and then it's ready.

**Comment by MongoCaleb** [comment]
@papafe can you please rebase from upstream/master? That should fix the build errors.

**Comment by papafe** [comment]
@MongoCaleb Merged from master and corrected according to the two notes

**Comment by MongoCaleb** [comment]
@papafe one more: we *never* change the files in /generated/. They are built from the test source. Can you please revert that change?

**Comment by MongoCaleb** [comment]
Superceded by https://github.com/mongodb/docs-realm/pull/2466, which fixes the issue with editing the generated file. Also allows us to test autobuilder tests before merging.

**Pull request closed without merging**